guidance/los_guidance/los_guidance/los_guidance.py

```diff
@@ -0,0 +1,49 @@
+import numpy as np
+
+class LOSGuidance:
+    def __init__(self, p0: list[float], p1: list[float], p_next: list[float]):
+        self.set_path(p0, p1)
+        self.heading_ref = 50*np.pi/180 # magic init number!!!
+        self.p_next = [np.array([50, -40]), np.array([20.0, -80.0]), np.array([120.0, -60.0]), np.array([160, 0.0]), np.array([60.0, 60.0])]
+        self.acceptance_radius = 0.5
+
+    def set_path(self, p0: list[float], p1: list[float]):
+        self.p0 = np.array(p0)
+        self.p1 = np.array(p1)
+
+    def calculate_R_Pi_p(self):
+        self.Pi_p = np.arctan2(self.p1[1]-self.p0[1], self.p1[0]-self.p0[0])
+        self.R_Pi_p = np.array(
+            [[np.cos(self.Pi_p), -np.sin(self.Pi_p)],
+            [np.sin(self.Pi_p), np.cos(self.Pi_p)]]
+        )
```

Comment:
Generally not super nice to have methods like these that modify internal state when the result is used right on computation. Something like
```python
    def calculate_R_Pi_p(self) -> np.ndarray:
        pi_p = np.arctan2(self.p1[1]-self.p0[1], self.p1[0]-self.p0[0])
        c = np.cos(pi_p)
        s = np.sin(pi_p)
       
        return np.array([[c, -s], [s, c]])
 ```
 lets you avoid maintaining Pi_p and R_Pi_p as internal state